### PR TITLE
Refactoring passenger config for easier maintenance and greater flexibility

### PR DESCRIPTION
**Pull request by bbergstrom**
Refactored how nginx `passenger.conf` is built so its flexibile for any of the many passenger configuration paremeters one might add.  Defaulting global queue to on but leaving the rest as is to be customized by the consuming cookbook.

### CODE REVIEW

File: attributes/default.rb [hunk 1]
```diff
@@ -41,8 +41,8 @@
 
 # Passenger
 default[:passenger][:version] = "3.0.21"
-default[:passenger][:root] = "/usr/local/lib/ruby/gems/1.9.1/gems/passenger-#{node[:passenger][:version]}"
-default[:passenger][:ruby] = node[:ruby_wrapper][:install_path]
+default[:passenger][:conf]["passenger_root"] = "/usr/local/lib/ruby/gems/1.9.1/gems/passenger-#{node[:passenger][:version]}"
```
Comment (anfleene) [inline attributes/default.rb]: Why did you have to convert these to strings?

File: templates/default/passenger.conf.erb [hunk 2]
```diff
@@ -1,27 +1,5 @@
...
-passenger_log_level <%= @config[:log_level] %>;
-
-<% unless @config[:default_user].nil? %>
-passenger_default_user <%= @config[:default_user] %>;
-<% end %>
-
-<% unless @config[:default_group].nil? %>
-passenger_default_group <%= @config[:default_group] %>;
+<% @passenger_conf.each do |name, value| %>
+<%= "#{name} #{value};" unless value.nil? %>
```
Comment (anfleene) [inline templates/default/passenger.conf.erb]: you could call `name.to_s` here so that you don't have to change the attributes

Comment (anfleene) [inline templates/default/passenger.conf.erb]: actually I'm pretty sure interpolating a symbol turns it into a string.

File: attributes/default.rb (hunk 1 above)
Comment (bbergstrom) [inline attributes/default.rb]: I thought they would `to_s` with a ":" prefix.

Comment (anfleene) [inline attributes/default.rb]: nope

Comment (anfleene) [inline attributes/default.rb]:
```ruby
[~/sportngin/guardhouse-firewall-manager]$ irb                                                                                                          [master][ruby-2.1.5@guardhouse-firewall-manager]
cannot load such file -- map_by_method
2.1.5 :001 > :derp.to_s
 => "derp"
2.1.5 :002 > "#{:derp}"
 => "derp"
```

**Comment by anfleene** [comment]
Changes look good :+1:

**Pull request merged**